src/math/Interpolate.cc

```diff
@@ -157,7 +157,7 @@ ::gsl_interp_type const *styleToGslInterpType(Interpolate::Style const style) {
         case Interpolate::UNKNOWN:
             throw LSST_EXCEPT(pex::exceptions::InvalidParameterError,
                               "I am unable to make an interpolator of type UNKNOWN");
-        case Interpolate::NUM_STYLES:
+        default:
```

Comment:
I turns out recent gcc can warn in this situation, but it requires a flag: https://stackoverflow.com/questions/5402745/gcc-switch-on-enum-retain-missing-warning-but-use-default